src/bin/cargo-plumbing/plumbing/locate_project.rs

```diff
@@ -0,0 +1,108 @@
+use std::fmt::Debug;
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use cargo_plumbing::CargoResult;
+use clap::ValueEnum;
+use serde::Serialize;
+use toml::Table;
+
+#[derive(Debug, Clone, Copy, ValueEnum)]
+enum MessageFormat {
+    Json,
+    Plain,
+}
```

Comment:
We will not be able to meet our plumbing needs through plain output.

Maybe one day we will have a `--format-version` flag like `cargo metadata` but that will be part of figuring out our schema evolution story.